### PR TITLE
Modified owners validator to accept dict

### LINKED ISSUE #827: [ELE-716] Report fails to generate if owner is configured as dict (exposures format)
**Describe the bug**
We expect a string or list of strings in the owner field for sources. 
For other format, we fail. 
Example: {"name":   "Maayan Salom", "email":   "maayan@elementary-data.com"}

**To Reproduce**
Configure source owner like this:
```
owner:
  name: name
  email: email@gmail.com
```

**Expected behavior**
We shouldn't crush on it, even if we don't support thr format. 

**Screenshots**
image

<sub>[ELE-716](https://linear.app/elementary/issue/ELE-716/report-fails-to-generate-if-owner-is-configured-as-dict-exposures)</sub>

### PR DESCRIPTION
**Pull request by manulpatel**
Fixes #827 

Modified `load_owners` validator to accept and return dict. This would prevent report failure.

### CODE REVIEW

File: elementary/utils/schema.py [hunk 1]
```diff
@@ -18,12 +18,15 @@ def _load_var_to_dict(var: Union[str, dict]) -> dict:
...
-        if not var:
+        if not var & type(var) != dict:
             return []
         elif isinstance(var, list):
             return var
         elif isinstance(var, str):
             loaded_var = try_load_json(var)
+            if type(loaded_var) is dict:
+                loaded_var = json.dumps(loaded_var)
+                return loaded_var
```
Comment (IDoneShaveIt) [inline elementary/utils/schema.py:29]:
```suggestion
                loaded_var = [json.dumps(loaded_var)]
```

File: elementary/utils/schema.py [hunk 2]
```diff
@@ -18,12 +18,15 @@ def _load_var_to_dict(var: Union[str, dict]) -> dict:
 
     @staticmethod
     def _load_var_to_list(var: Union[str, list]) -> list:
-        if not var:
+        if not var & type(var) != dict:
```
Comment (IDoneShaveIt) [inline elementary/utils/schema.py:21]:
```suggestion
        if not var:
```
I don't think this one should be changed.
Currently we support Union[str, list] so if someone is passing a dict it should be OK to fail 🙂.

**Review by IDoneShaveIt — COMMENTED** [review]
Thanks for the contribution!
Left some small comments 🙂

File: elementary/utils/schema.py [hunk 3]
```diff
@@ -18,12 +18,15 @@ def _load_var_to_dict(var: Union[str, dict]) -> dict:
...
             return []
         elif isinstance(var, list):
             return var
         elif isinstance(var, str):
             loaded_var = try_load_json(var)
+            if type(loaded_var) is dict:
```
[the remark this answers is not in the document]
Comment (manulpatel) [reply elementary/utils/schema.py:27]: Noted

**Review by IDoneShaveIt — APPROVED** [review]
Looking good!
Thank you for the contribution 💪🏻 💪🏻

**Pull request merged**